modules/tinymce/src/plugins/advlist/main/ts/ui/Buttons.ts

```diff
@@ -80,21 +83,23 @@ const addSplitButton = (editor: Editor, id: string, tooltip: string, cmd: string
   });
 };
 
-const addButton = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, _styles: string[]): void => {
+const addButton = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, styleValue: string): void => {
+  const value = normalizeStyleValue(styleValue);
   editor.ui.registry.addToggleButton(id, {
     active: false,
     tooltip,
     icon: nodeName === ListType.OrderedList ? 'ordered-list' : 'unordered-list',
     onSetup: makeSetupHandler(editor, nodeName),
-    onAction: () => editor.execCommand(cmd)
+    // Need to make sure the button removes rather than applies if a list of the same type is selected
+    onAction: () => editor.queryCommandState(cmd) || value === '' ? editor.execCommand(cmd) : Actions.applyListFormat(editor, nodeName, value)
   });
 };
 
 const addControl = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, styles: string[]): void => {
   if (styles.length > 1) {
     addSplitButton(editor, id, tooltip, cmd, nodeName, styles);
   } else {
-    addButton(editor, id, tooltip, cmd, nodeName, styles);
+    addButton(editor, id, tooltip, cmd, nodeName, styles[0] || '');
```

Comment:
Would it be better to call normalizeStyleValue on the first style[] element here? And replace the empty string with 'default' so that it is known an empty string will be returned?

Reply:
I am not quite sure if I understand what you are suggesting so I would probably need to see a code suggestion.
In saying that, maybe just doing `addButton(editor, id, tooltip, cmd, nodeName, styles[0] || 'default');` would be slightly better/easier to reason about?

Reply:
```suggestion
const addButton = (editor: Editor, id: string, tooltip: string, cmd: string, nodeName: ListType, value: string): void => {
```
```suggestion
    addButton(editor, id, tooltip, cmd, nodeName, normalizeStyleValue(styles[0]));
```
I suppose I was meaning something like this, removing the `const value = normalizeStyleValue(styleValue);` from `addButton`, but I also like your suggestion, so up to you.

Reply:
BTW I messed up the position of the suggestions, still getting the hang of it, so don't commit these ones haha

Reply:
I thought that might be what you were referring to for a suggestion. I would have to make a small change to `normalizeStyleValue` but it should work 👍

Reply:
I have pushed up a change more in line with what you were thinking. Thanks for the suggestion